### PR TITLE
Resolve /decide endpoint errors due to invalid token

### LINKED ISSUE #4712: /decide endpoint failure due to test tokens
Sentry Issue: [POSTHOG-2V8](https://sentry.io/organizations/posthog/issues/2411200673/?referrer=github_integration)

```
AttributeError: 'NoneType' object has no attribute 'startswith'
(1 additional frame(s) were not displayed)
...
  File "django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "newrelic/hooks/framework_django.py", line 563, in wrapper
    return wrapped(*args, **kwargs)
  File "django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "posthog/api/decide.py", line 92, in get_decide
    token, is_test_environment = _clean_token(token)
  File "posthog/api/capture.py", line 103, in _clean_token
    is_test_environment = token.startswith("test_")
```

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Fixes https://github.com/PostHog/posthog/issues/4712

The problem was just a missing check for token being None, but I refactored this code a bit to ensure we don't have to remember to check this, if we use `get_token` anywhere in the future.

## Checklist

- [x] All querysets/queries filter by Organization, by Team, and by User
- [x] Django backend tests
- [x] Jest frontend tests
- [x] Cypress end-to-end tests
- [x] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [x] Frontend/CSS is usable at 320px (iPhone SE) and decent at 360px (most phones)
- [x] Breaking changes are backwards-compatible. Ensure old/new frontend requests work with new/old backends, and vice versa.

### CODE REVIEW

File: posthog/api/capture.py [hunk 1]
```diff
@@ -98,11 +105,11 @@ def _get_token(data, request) -> Optional[str]:
     return None
 
 
-# Support test_[apiKey] for users with multiple environments
-def _clean_token(token):
-    is_test_environment = token.startswith("test_")
-    token = token[5:] if is_test_environment else token
-    return token, is_test_environment
+def get_token(data, request) -> Tuple[Optional[str], bool]:
```
Comment (macobo) [inline posthog/api/capture.py:108]: It's sort of weird we have both `_get_token` and `get_token` functions, I can see myself confused. How about refactoring the existing `_get_token` function?

**Review by macobo — APPROVED** [review]
Nice, clean fix. Good job!

**Comment by neilkakkar** [comment]
Failing Cypress test seems flakey. Re-running again to see if it passes _soon_.

**Review by buwilliams — APPROVED** [review]
lgtm

**Pull request merged**